comp/dogstatsd/server/component.go

```diff
@@ -0,0 +1,67 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2016-present Datadog, Inc.
+
+// Package server implements a component to run the dogstatsd server
+package server
+
+import (
+	"time"
+
+	"github.com/DataDog/datadog-agent/pkg/aggregator"
+	"github.com/DataDog/datadog-agent/pkg/dogstatsd"
+	"github.com/DataDog/datadog-agent/pkg/util/fxutil"
+	"go.uber.org/fx"
+)
+
+// team: Agent Metrics Logs
+
+// Component is the component type.
+type Component interface {
+
+	// Start starts the dogstatsd server
+	Start(demultiplexer aggregator.Demultiplexer) error
+	// Stop stops the dogstatsd server
+	Stop()
+
+	// Capture starts a new dogstatsd traffic capture
+	Capture(p string, d time.Duration, compressed bool) error
+
+	// IsCaputreOngoing returns true if a capture is ongoing
+	IsCaputreOngoing() bool
+
+	// GetCapturePath returns the capture destination path
+	GetCapturePath() (string, error)
```

Comment:
Based on https://github.com/DataDog/datadog-agent/blob/aeb058147baa26e9c7c4708ab46f16512dc34ad0/cmd/agent/api/grpc.go#L79-L89 you can have a single method for `Capture`.